src/setfit/modeling.py

```diff
@@ -13,17 +13,77 @@
 from sklearn.linear_model import LogisticRegression
 from sklearn.multiclass import OneVsRestClassifier
 from sklearn.multioutput import ClassifierChain, MultiOutputClassifier
+from torch.utils.data import DataLoader, Dataset
+from tqdm import tqdm
+from pathlib import Path
 
 from . import logging
 
 
+if TYPE_CHECKING:
+    from numpy import ndarray
+    from transformers import PreTrainedTokenizerBase
+
+
 logging.set_verbosity_info()
 logger = logging.get_logger(__name__)
 
-
+TokenizerOutput = Dict[str, List[int]]
 MODEL_HEAD_NAME = "model_head.pkl"
 
 
+class SetFitDataset(Dataset):
```

Comment:
I think this would belong better in `setfit.data` - WDYT?

Could we also add a one-line docstring to summarise what it's for?

Reply:
That's a good idea! Will also add doctoring as well

Reply:
done!